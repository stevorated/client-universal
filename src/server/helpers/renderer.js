import React from 'react'
import { renderToString } from 'react-dom/server'
import { Helmet } from 'react-helmet'
import { StaticRouter } from 'react-router'
import { Provider } from 'react-redux'
import { ApolloProvider } from 'react-apollo'
import serialize from 'serialize-javascript'
import { ServerStyleSheet } from 'styled-components'
import Layout from '../../shared/Routes/Layout'
import { config } from '@fortawesome/fontawesome-svg-core'
config.autoAddCss = false

const { API_BASE } = process.env
const linkToFavicon = `${API_BASE}/public_images/favicon.png`
export default function renderer(req, store, client, context) {

  const App = (
    <Provider store={store}>
      <ApolloProvider client={client}>
        <StaticRouter location={req.url} context={context}>
          <Layout />
        </StaticRouter>
      </ApolloProvider>
    </Provider>
  )

  const sheet = new ServerStyleSheet()
  const content = renderToString(sheet.collectStyles(App))
  const styles = sheet.getStyleTags()
  
  const helmet = Helmet.renderStatic()

  return `
  <!DOCTYPE html>
    <html>
      <head>
      <meta charset="UTF-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      <meta name="840430c5e70010722d679edee63ea147c19430e1" content="e73a6ac6dc339154d6038998ef2574d65b12bf5e" />
      <script type="text/javascript">
        WebFontConfig = {
          google: { families: [ 'Sigmar One::latin' ] }
        };
        (function() {
          var wf = document.createElement('script');
          wf.src = ('https:' == document.location.protocol ? 'https' : 'http') +
            '://ajax.googleapis.com/ajax/libs/webfont/1/webfont.js';
          wf.type = 'text/javascript';
          wf.async = 'true';
          var s = document.getElementsByTagName('script')[0];
          s.parentNode.insertBefore(wf, s);
        })(); </script>
      <link href="https://fonts.googleapis.com/css?family=Sigmar+One&display=swap" rel="stylesheet" />
      <link href="https://use.fontawesome.com/releases/v5.9.0/css/svg-with-js.css" rel="stylesheet" />
      ${helmet.title.toString()}
      ${helmet.meta.toString()}
      <link rel="icon" href="${linkToFavicon}" type="image/png"/>
      <link rel="stylesheet" href="/styles/main.css"/>
      ${styles}
      </head>
      <body>
        <div id="root">${content}</div>
        <div id="portal"></div>
        <script>
          window.__INITIAL_STATE__ = ${serialize(store.getState())}
        </script>
        <script>
          window.__APOLLO_STATE__ = ${serialize(client.extract())}
        </script>
        <script src="/bundle.js"></script>
      </body>
    </html>
  
  `
}
