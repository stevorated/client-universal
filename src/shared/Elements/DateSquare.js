import React, { Fragment } from 'react'

export const DateSquare =  ({ month = 'Jan', day ='2' }) => {
  return (
    <div>
      <div
        data-test="month"
        className="noPadding"
        style={{fontSize: '1.2rem'}}
        >
        {month}
      </div>
      <hr className="noPadding"/>
      <div
        data-test="day"
        style={{fontSize: '2rem'}}
        >
        {day}
      </div>
    </div>
  )
}

export default DateSquare