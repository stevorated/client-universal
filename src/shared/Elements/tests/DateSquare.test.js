import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { DateSquare } from '../DateSquare'

import 'jest-styled-components'

describe('<DateSquare />', () => {
  it('matches Snapshot', ()=> {
    const tree = renderer.create(<DateSquare />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders with correct defaults', ()=> {
    const wrapper = shallow(<DateSquare />)
    const day = wrapper.find(`[data-test="day"]`)
    const month = wrapper.find(`[data-test="month"]`)
    expect(month.text()).toEqual('Jan')
    expect(month.length).toBe(1)
    expect(day.text()).toEqual('2')
    expect(day.length).toBe(1)
  })
})
