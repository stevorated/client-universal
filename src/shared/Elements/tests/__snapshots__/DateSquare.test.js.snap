// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateSquare /> matches Snapshot 1`] = `
<div>
  <div
    className="noPadding"
    data-test="month"
    style={
      Object {
        "fontSize": "1.2rem",
      }
    }
  >
    Jan
  </div>
  <hr
    className="noPadding"
  />
  <div
    data-test="day"
    style={
      Object {
        "fontSize": "2rem",
      }
    }
  >
    2
  </div>
</div>
`;
